src/Decompiler/Evaluation/ExpressionSimplifier.cs

```diff
@@ -1163,6 +1172,33 @@ public virtual (Expression, bool) VisitMkSequence(MkSequence seq)
             }
         }
 
+        private Expression? FuseSequenceOfSlicedMultiplications(Expression[] seq)
+        {
+            BinaryExpression? mul = null;
+            for (int i = seq.Length -1; i >= 0; --i)
+            {
+                if (seq[i] is Slice slice && 
+                    slice.Expression is BinaryExpression bin && 
+                    bin.Operator is IMulOperator)
+                {
+                    if (mul is null)
+                        mul = bin;
+                    else if (!cmp.Equals(mul.Left, bin.Left) ||
+                            !cmp.Equals(mul.Right, bin.Right))
+                        return null;
+                    else
+                        mul = bin;
```

Comment:
Looks like incorrect logic for me. If sequence contains any slice with multiplication operator then this code discards other elements of sequence.

Reply:
Now if the sequence contains anything else than multiplications or slices of that same multiplication, the transformation will not occur.